packages/gatsby-plugin-screenshot/src/lambda/index.js

```diff
@@ -0,0 +1,159 @@
+const setup = require(`./starter-kit/setup`)
+
+const crypto = require(`crypto`)
+const fs = require(`fs`)
+
+const tmp = require(`tmp`)
+const AWS = require(`aws-sdk`)
+const s3 = new AWS.S3({
+  apiVersion: `2006-03-01`,
+})
+
+exports.handler = async (event, context, callback) => {
+  // For keeping the browser launch
+  context.callbackWaitsForEmptyEventLoop = false
+
+  const url = event.url
+
+  if (!url) {
+    callback(`no url provided`)
+    return
+  }
+
+  const width = event.width || 1024
+  const height = event.height || 768
+
```

Comment:
Pardon my ignorance, looked at the PR for 5 minutes only and still have not a real clue what is going on 😉 – but wondering if it would be possible to define a custom screenshot size/browser window size, e.g. to get screenshots reflecting the "mobile layout" of a website?

Reply:
This is a great point! At the moment, I did make the lambda accept `width` and `height` options, but the plugin doesn't specify them. It'd be easy to add, though.

Properly rendering mobile layout may also require setting some of the other arguments to puppeteer's `setViewport` method, such as `isMobile` and `isLandscape`: https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#pagesetviewportviewport.